contracts/FountainV1.sol

```diff
@@ -450,42 +449,54 @@ contract FountainV1 {
         }
 
         // Cannot update active moneyPool, check if there is a pending moneyPool
-        moneyPoolId = _getPendingMoneyPoolId(who);
-        if (moneyPoolId != 0) {
-            return moneyPoolId;
-        }
+        moneyPoolId = _getPendingMoneyPoolId(owner);
+        if (moneyPoolId != 0) return moneyPoolId;
 
         // No pending moneyPool found, clone the latest moneyPool
-        moneyPoolId = _getLatestMoneyPoolId(who);
-        return _createMoneyPoolFromId(moneyPoolId, now);
+        moneyPoolId = _getLatestMoneyPoolId(owner);
+
+        if (moneyPoolId != 0) return _createMoneyPoolFromId(moneyPoolId, now);
+
+        moneyPoolId = _initMoneyPoolId(owner);
+        MoneyPool storage moneyPool = moneyPools[moneyPoolId];
+        moneyPool.start = now;
+
+        return moneyPoolId;
     }
 
     /// @dev Only active MoneyPools can be sustained.
-    /// @param who The address to find a MoneyPool for.
+    /// @param owner The address who owns the MoneyPool to look for.
     /// @return id The resulting id.
-    function _moneyPoolIdToSustain(address who) private returns (uint256 id) {
+    function _moneyPoolIdToSustain(address owner) private returns (uint256 id) {
         // Check if there is an active moneyPool
-        uint256 moneyPoolId = _getActiveMoneyPoolId(who);
-        if (moneyPoolId != 0) {
-            return moneyPoolId;
-        }
+        uint256 moneyPoolId = _getActiveMoneyPoolId(owner);
+        if (moneyPoolId != 0) return moneyPoolId;
 
         // No active moneyPool found, check if there is a pending moneyPool
-        moneyPoolId = _getPendingMoneyPoolId(who);
+        moneyPoolId = _getPendingMoneyPoolId(owner);
         if (moneyPoolId != 0) {
             return moneyPoolId;
         }
 
         // No pending moneyPool found, clone the latest moneyPool
-        moneyPoolId = _getLatestMoneyPoolId(who);
+        moneyPoolId = _getLatestMoneyPoolId(owner);
+
+        require(
+            moneyPoolId > 0,
+            "Fountain::moneyPoolIdToSustain: MoneyPool not found"
+        );
+
         MoneyPool storage latestMoneyPool = moneyPools[moneyPoolId];
         // Use a start date that's a multiple of the duration.
         // This creates the effect that there have been scheduled MoneyPools ever since the `latest`, even if `latest` is a long time in the past.
         uint256 start = _determineModuloStart(
             latestMoneyPool.start.add(latestMoneyPool.duration),
             latestMoneyPool.duration
         );
-        return _createMoneyPoolFromId(moneyPoolId, start);
+
+        uint256 newMoneyPoolId = _createMoneyPoolFromId(moneyPoolId, start);
```

Comment:
Is there a reason to assign to a variable and then return it instead of just returning the function call result?

Reply:
good eye. nope, there was an event emitted before that I removed.